app/assets/javascripts/govuk_publishing_components/components/checkboxes.js

```diff
@@ -5,90 +5,118 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}
+        if (category) {
+          var uncheckTrackCategory = $checkbox.getAttribute('data-uncheck-track-category')
+          if (!$checkbox.checked && uncheckTrackCategory) {
+            category = uncheckTrackCategory
           }
+          var action = $checkbox.getAttribute('data-track-action')
+          var options = $checkbox.getAttribute('data-track-options')
+          if (options) {
+            options = JSON.parse(options)
+          } else {
+            options = {}
+          }
+          options.value = $checkbox.getAttribute('data-track-value')
+          options.label = $checkbox.getAttribute('data-track-label')
+          window.GOVUK.analytics.trackEvent(category, action, options)
         }
-      })
-
-      $(scope).on('change', '[data-exclusive=true] input[type=checkbox]', function (e) {
-        var currentCheckbox = e.target
-        var checkboxes = currentCheckbox.closest('.govuk-checkboxes')
-        var exclusiveOption = $(checkboxes).find('input[type=checkbox][data-exclusive]')
-        var nonExclusiveOptions = $(checkboxes).find('input[type=checkbox]:not([data-exclusive])')
-
-        if (currentCheckbox.dataset.exclusive === 'true' && currentCheckbox.checked === true) {
-          nonExclusiveOptions.each(function () {
-            $(this).prop('checked', false)
-          })
-        } else if (currentCheckbox.dataset.exclusive !== 'true' && currentCheckbox.checked === true) {
-          exclusiveOption.prop('checked', false)
-        }
-      })
+      }
     }
+  }
 
-    this.toggleNestedCheckboxes = function (scope, checkbox) {
-      if (checkbox.checked) {
-        scope.find('input[type=checkbox]').prop('checked', true)
-      } else {
-        scope.find('input[type=checkbox]').prop('checked', false)
+  GovukCheckboxes.prototype.handleNestedCheckboxChange = function (event) {
+    var $checkbox = event.target
+    var $isNested = $checkbox.closest('.govuk-checkboxes--nested')
+    var $hasNested = this.$module.querySelector('.govuk-checkboxes--nested[data-parent=' + $checkbox.id + ']')
+
+    if ($hasNested) {
+      this.toggleNestedCheckboxes($hasNested, $checkbox)
+    } else if ($isNested) {
+      this.toggleParentCheckbox($isNested, $checkbox)
+    }
+  }
+
+  GovukCheckboxes.prototype.toggleNestedCheckboxes = function ($scope, $checkbox) {
+    var $nestedCheckboxes = $scope.querySelectorAll('input[type=checkbox]')
+    if ($checkbox.checked) {
+      for (var i = 0; i < $nestedCheckboxes.length; i++) {
+        $nestedCheckboxes[i].checked = true
       }
+    } else {
+      for (i = 0; i < $nestedCheckboxes.length; i++) {
+        $nestedCheckboxes[i].checked = false
+      }
+    }
+  }
+
+  GovukCheckboxes.prototype.toggleParentCheckbox = function ($scope, $checkbox) {
+    var $inputs = $scope.querySelectorAll('input')
+    var $checkedInputs = $scope.querySelectorAll('input:checked')
+    var parentId = $scope.dataset.parent
```

Comment:
Could `dataset` be replaced with the more compatible `getAttribute()`?
```suggestion
    var parentId = $scope. getAttribute('data-parent')
```

Reply:
Sorry about the typo!